packages/tools/package/src/models/package/package.ts

```diff
@@ -93,6 +93,40 @@ class Package {
       .then(() => this);
   }
 
+  /**
+   * Apply the modifications of this Package instance to the package definition
+   * files.
+   *
+   * @returns - Promse that resolves to this Package instance.
+   */
+  public syncDependency(deps: Record<string, string>, devDeps: boolean = false): Promise<this> {
+    const packageDefinitionPath = path.join(this.data.location, CONSTANTS.PACKAGE_DEFINITION_FILE);
+
+    return Package.readDefinition({ definitionPath: packageDefinitionPath })
+      .then((definition) => {
+        const dependencies = (devDeps) ? definition.devDependencies : definition.dependencies;
+
+        Object.entries(deps).map(([name, version]) => {
```

Comment:
i am little confused on how this work can we add a comment here

Reply:
Will take care of this in the next commit. I'll add a remark in the function description
Basically we iterate over the dependency map in the argument which arrives from npm and overwrite the dependencies in the package.json

Reply:
@arun3528 I've removed this function as it is not needed.